ospf6d/ospf6_interface.c

```diff
@@ -388,6 +389,22 @@ ospf6_interface_state_update (struct interface *ifp)
   if (CHECK_FLAG (oi->flag, OSPF6_INTERFACE_DISABLE))
     return;
 
+  /* Adjust the mtu values if the kernel told us something new */
+  if (ifp->mtu6 != oi->ifmtu)
+    {
+      /* If nothing configured, just accept it */
+      if (!oi->c_ifmtu)
+        oi->ifmtu = ifp->mtu6;
```

Comment:
There is a potential problem to fix the oi->ifmtu to ifp->mtu6 if the oi->ifmtu has been reduce to the I/O buffer size. Look at line 206-215 of ospf6_interface.c / ospf6_interface_create() function. The oi->ifmtu will become larger than the I/O buffer. I think it will be safer to adjust the I/O buffer and eventually MTU like it is done in ospf6_interface_create() function by calling ospf6_iobuf_size(ifp->mtu6) at the end of your modification.